From 390e052af47bd792592147b6def4071904d6c653 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Wed, 1 Jun 2022 17:27:08 +0200
Subject: [PATCH] boot src cmd: custom scan_overlays cmd

Add possibility to specify the DTO to apply using "overlay" u-boot
env var.
If the variable is not empty, content of /overlays/overlays.txt is
ignored. Otherwise the DTO to apply come from /overlays/overlays.txt.
---
 boot.scr.cmd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boot.scr.cmd
index f6328eb..d695a7a 100644
--- a/boot.scr.cmd
+++ b/boot.scr.cmd
@@ -25,7 +25,7 @@ env set scan_m4fw 'if test -e ${devtype} ${devnum}:${distro_bootpart} ${m4fw_nam
 # management of overlay
 env set ov_init 'load ${devtype} ${devnum}:${distro_bootpart} ${fdt_addr_r} ${fdtfile} && env set fdt_addr ${fdt_addr_r} && fdt addr ${fdt_addr} && setexpr fdtovaddr ${fdt_addr} + C0000'
 env set ov_apply 'test -n ${fdtovaddr} && test -n ${overlay} && for ov in ${overlay}; do echo overlaying ${ov}...; load ${devtype} ${devnum}:${distro_bootpart} ${fdtovaddr} /overlays/${ov}.dtbo && fdt resize ${filesize} && fdt apply ${fdtovaddr}; done'
-env set scan_overlays 'if test -e ${devtype} ${devnum}:${distro_bootpart} /overlays/overlays.txt && load ${devtype} ${devnum}:${distro_bootpart} ${loadaddr} /overlays/overlays.txt && env import -t ${loadaddr} ${filesize}; then echo loaded overlay.txt: ${overlay}; run ov_init; run ov_apply; fi'
+env set scan_overlays 'if test -z "${overlay}"; then test -e ${devtype} ${devnum}:${distro_bootpart} /overlays/overlays.txt && load ${devtype} ${devnum}:${distro_bootpart} ${loadaddr} /overlays/overlays.txt && env import -t ${loadaddr} ${filesize}; fi; if test -n ${overlay}; then echo loaded overlay: ${overlay}; run ov_init; run ov_apply; fi'
 
 # Update the DISTRO command to search in sub-directory and load M4 firmware
 env set boot_prefixes "/${boot_device}${boot_instance}_"
-- 
2.25.1

